### ISSUE TITLE
MalwarePatrol malwarepatrol.ndb database integrity tested BAD

### ISSUE THREAD

**Issue by Th-Barth** [extremeshok/clamav-unofficial-sigs#255]
Since 31. August clamav-unofficials tries to update the malwarepatrol file, but clamscan always reports bad integrity test.

/var/log/clamav-unofficial-sigs/clamav-unofficial-sigs.log
Sep 14 23:54:41 MalwarePatrol  Database File Update
Sep 14 23:54:44 Testing updated MalwarePatrol database file: malwarepatrol.ndb
Sep 14 23:54:44 Clamscan reports MalwarePatrol malwarepatrol.ndb database integrity tested BAD 
Removed invalid database: /var/lib/clamav-unofficial-sigs/dbs-mbl/malwarepatrol.ndb

ls -al /var/lib/clamav-unofficial-sigs/dbs-mbl/
total 12
drwxr-xr-x  2 clamav clamav 4096 Sep 17 23:49  .
drwxr-xr-x 11 clamav clamav 4096 Aug 12 15:01  ..
lrwxrwxrwx  1 clamav clamav   21 Aug 12 15:20 'getfile?product=xx&list=clamav_ext&receipt=xxx&hash=1' -> malwarepatrol.ndb.md5
-rw-r--r--  1 clamav clamav   33 Sep 17 23:49  malwarepatrol.ndb.md5

Malware support told me that everything is ok with my account. Basic Defense contains the last 15 days of active malware URLs from their database, updated every 4 hours. This is new since the beginning of the year.

This is the old malwarepatrol.ndb
ls -al /var/lib/clamav/malwarepatrol.ndb 
-rw-r--r-- 1 clamav clamav 25974461 Aug 26 22:06 /var/lib/clamav/malwarepatrol.ndb

When I download the signature db with the MalwarePatrolDownload.sh script from MalwarePatrol I get a file  like this

ls -al /opt/malwarepatrol-ext.db
-rw-r--r-- 1 root root 64690 Sep 15 18:53 /opt/malwarepatrol-ext.db
The format of the content of this files is
MBL_36722379=64656c6c312e75672f66696c65732f70656e656c6f70

Just 64 KB size, containing the malware urls of the last 15 days only.

# clamscan -d /opt/malwarepatrol-ext.db scan-test.txt
scan-test.txt: Empty file

----------- SCAN SUMMARY -----------
Known viruses: 825
Engine version: 0.101.4
Scanned directories: 0
Scanned files: 0
Infected files: 0
Data scanned: 0.00 MB
Data read: 0.00 MB (ratio 0.00:1)
Time: 0.012 sec (0 m 0 s)

The format of the old ndb file is:
MBL_10000346:0:*:736f6674646c2e333630747063646e2e636f6d2f6b75676f75

Any idea how to get a good integrity test finally? Unfortunately, I have to cancel my subsription at malwarepatrol if there is no solution for this problem.

**Comment by extremeshok** [comment]
Please try the latest dev version

**Comment by Th-Barth** [comment]
Sorry, same problem. May be I should prepare the MalwarePatrolDownload.sh script  for a 4 hour download and just move the db files into /var/lib/clamav

Sep 20 19:48:26 Checking for MalwarePatrol updates...
Sep 20 19:48:26 MalwarePatrol  Database File Update
Sep 20 19:48:28 Testing updated MalwarePatrol database file: malwarepatrol.ndb
Sep 20 19:48:28 Clamscan reports MalwarePatrol malwarepatrol.ndb database integrity tested BAD
Sep 20 19:48:28 Removed invalid database: /var/lib/clamav-unofficial-sigs/dbs-mbl/malwarepatrol.ndb
Sep 20 19:48:28 No updates detected, ClamAV databases were not reloaded
Sep 20 19:48:28 Issue tracker : https://github.com/extremeshok/clamav-unofficial-sigs/issues
Sep 20 19:48:29       Powered By https://eXtremeSHOK.com      
Sep 20 19:55:04 
Sep 20 19:55:04 *** SCRIPT INFORMATION ***
Sep 20 19:55:04 clamav-unofficial-sigs.sh 6.3.0 (2019-09-02)
Sep 20 19:55:04 Master.conf Version: 81
Sep 20 19:55:04 Minimum required config: 80
Sep 20 19:55:04 *** SYSTEM INFORMATION ***
Sep 20 19:55:04 *** CLAMSCAN LOCATION & VERSION ***
Sep 20 19:55:04 /usr/bin/clamscan
Sep 20 19:55:04 *** RSYNC LOCATION & VERSION ***
Sep 20 19:55:04 /usr/bin/rsync
Sep 20 19:55:04 *** CURL LOCATION & VERSION ***
Sep 20 19:55:04 /usr/bin/curl
Sep 20 19:55:04 *** GPG LOCATION & VERSION ***
Sep 20 19:55:04 /usr/bin/gpg
Sep 20 19:55:04 *** DIRECTORY INFORMATION ***
Sep 20 19:55:04 Working Directory: /var/lib/clamav-unofficial-sigs
Sep 20 19:55:04 Clam Database Directory: /var/lib/clamav
Sep 20 19:55:04 Configuration Directory: /etc/clamav-unofficial-sigs

**Comment by zeusdesignco** [comment]
Hi, I am having an issue with the malwarepatrol db as well. I am running centos6, have the config file set up properly. I get this error when running the script:

/usr/local/bin/clamav-unofficial-sigs.sh: line 2828: /var/lib/clamav-unofficial-sigs/test/malwarepatrol.db: No such file or directory
LibClamAV Error: cl_load(): No such file or directory: /var/lib/clamav-unofficial-sigs/test/malwarepatrol.db
ERROR: Can't get file status
/bin/grep: /var/lib/clamav-unofficial-sigs/test/malwarepatrol.db: No such file or directory
/usr/local/bin/clamav-unofficial-sigs.sh: line 2835: /var/lib/clamav-unofficial-sigs/test/malwarepatrol.db-tmp: No such file or directory
mv: cannot stat `/var/lib/clamav-unofficial-sigs/test/malwarepatrol.db-tmp': No such file or directory
/usr/local/bin/clamav-unofficial-sigs.sh: line 2917: 15388 Aborted                 $clamscan_bin --quiet -d "${work_dir_yararulesproject}/${db_file}" "${work_dir_work_configs}/scan-test.txt" 2> /dev/null

I don't see a test directory anywhere, why isn't it there?

Thanks for your help.

**Comment by msapiro** [comment]
I have the same issue with Version 6.1.1 at commit 51c90643b1e. I worked around it with this patch:
```
--- clamav-unofficial-sigs.sh	2019-11-23 13:57:30.668177314 -0800
+++ /usr/local/sbin/clamav-unofficial-sigs	2019-11-23 15:12:41.224220704 -0800
@@ -2736,7 +2736,7 @@
       else
         last_malwarepatrol_update="0"
       fi
-      db_file=""
+      db_file="$malwarepatrol_db"
       update_interval="$((malwarepatrol_update_hours * 3600))"
       time_interval="$((current_time - last_malwarepatrol_update))"
       if [ "$time_interval" -ge "$((update_interval - 600))" ] ; then
@@ -2761,9 +2761,13 @@
           ret="$?"
           if [ "$ret" -eq 0 ] ; then
             if cmp -s "${work_dir_malwarepatrol}/${malwarepatrol_db}" "${clam_dbs}/${malwarepatrol_db}" ; then
-              malwarepatrol_reloaded=1
+              malwarepatrol_reloaded=0
             else
-              malwarepatrol_reloaded=2
+              if [ -n "$ham_dir" ] ; then
+                malwarepatrol_reloaded=2
+              else
+                malwarepatrol_reloaded=1
+              fi
             fi
           else # Wget failed
             malwarepatrol_reloaded=-1
```
I do not believe this is a proper fix, but it does work around the issue as long as ham_dir is not set in the configs. There is some other issue with the whole ham_dir and testing process.

**Comment by extremeshok** [comment]
Any chance you have a working solution ?

**Comment by m0urs** [comment]
For me it does work with these settings:

malwarepatrol_receipt_code="<your receipt code from the URL>"
malwarepatrol_product_code="15"

malwarepatrol_list="clamav_basic" # clamav_basic or clamav_ext
malwarepatrol_free="no"

I first used clamav_ext with which I also did get that integrity failed message. With the settings above, everything is working now without the need to apply that fix above.

**Comment by extremeshok** [comment]
part of #274

**Comment by msapiro** [comment]
Agreed that this is fixed in 6.3.0.
